frontend/src/components/Layout/AppBar/Buttons/TaskManagerButton.vue

```diff
@@ -68,128 +61,113 @@ interface TaskInfo {
   id: string;
 }
 
-export default defineComponent({
-  props: {
-    fab: {
-      type: Boolean,
-      required: false
-    },
-    nudgeBottom: {
-      type: Number,
-      default: 5
-    },
-    timeout: {
-      type: Number,
-      default: 5000
-    }
-  },
-  setup() {
-    const taskManager = taskManagerStore();
+const props = withDefaults(
+  defineProps<{
+    fab?: boolean;
+    timeout: number;
+  }>(),
+  { timeout: 5000 }
+);
 
-    return { taskManager };
-  },
-  data() {
-    return {
-      menu: false,
-      showButton: false,
-      taskList: [] as TaskInfo[],
-      scheduledTimeout: 0
-    };
-  },
-  computed: {
-    buttonColor(): string | undefined {
-      return this.taskList.every((task) => {
-        return task.progress === 100;
-      })
-        ? 'success'
-        : undefined;
-    }
-  },
-  watch: {
-    'taskManager.tasks': {
-      immediate: true,
-      handler(): void {
-        this.getTaskList();
-
-        if (this.taskList.length > 0) {
-          this.showButton = true;
-        }
-      }
-    },
-    /**
-     * Handles all tasks completion
-     */
-    buttonColor() {
-      window.clearTimeout(this.scheduledTimeout);
-
-      if (this.buttonColor && !this.menu) {
-        this.setTimeout();
+const menu = ref(false);
+const taskList = ref<TaskInfo[]>([]);
+const scheduledTimeout = ref(0);
+const taskManager = taskManagerStore();
+
+const buttonColor = computed(() => {
+  return taskList.value.every((task) => {
+    return task.progress === 100;
+  })
+    ? 'success'
+    : undefined;
+});
+
+const showButton = computed(() => taskList.value.length > 0);
+
+/**
+ *  Clear the tasks list
+ */
+function clearState(): void {
+  taskList.value = [];
+}
+
+/**
+ * Set a timeout to hide the button
+ */
+function setTimeout(): void {
+  scheduledTimeout.value = window.setTimeout(clearState, props.timeout);
+}
+
+/**
+ * Locally retrieve and parse the tasks list
+ */
+function getTaskList(): void {
```

Comment:
Maybe this can be a computed property instead?

Reply:
I let it that way cause there's a timeout that clears it after 5s or so, and I wasn't sure how it interacted with the task state. So I preferred to keep it that way for the migration.